prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -161,6 +165,32 @@ def canonicalize_dataset_using_columns_for_split(
             {"input_col": input_col, "output_col": output_col}
         )
 
+    @staticmethod
+    def get_input_output_columns(
+        instruction, dataset_name, dataset_columns, example_rows
+    ):
+
+        prompt = construct_prompt_for_column_selection(
+            instruction, dataset_name, dataset_columns, example_rows
+        )
+        required_keys = ["input", "output"]
+        optional_keys = ["ambiguous", "irrelevant"]
+        response = parse_prompt_to_fields(prompt, required_keys, optional_keys)
+        input_columns = response["input"]
+        output_column = response["output"]
+        incorrect_columns = [col for col in input_columns if col not in dataset_columns]
+        incorrect_columns += [
+            col for col in output_column if col not in dataset_columns
+        ]
+        if (
+            len(incorrect_columns) > 0
+            or len(input_columns) < 1
+            or len(output_column) != 1
+        ):
+            raise ValueError("incorrect number of columns")
```

Comment:
This error should be clearer. I would say only one of these cases is an error: if `len(incorrect_columns) > 0` (in which case we can consider retrying the query).

`len(input_columns) < 1` and `len(output_column) != 1` are rather signals that this dataset is not usable, in which case we should just ignore this dataset entirely

Reply:
Hmm, so a different error for unusable dataset. And should I do any error handling in that case or should that be a separate issue?